### PR TITLE
[backport] Added X509 certificate context key when client certificate is present and pem trust store configuration

### PR DESCRIPTION
**Pull request by astromechza**
Backporting https://github.com/oracle/helidon/pull/4185#event-6613756686 to helidon 2.x branch with the intention to release it in 2.5.1.

### CODE REVIEW

File: webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        rawConfig.put("private-key.pem.cert-chain.resource.resource-path", "ssl/certificate.pem");
+
+        webServer = WebServer.builder(
+                Routing.builder()
+                        .any((req, res) -> {
+                            // This is annoyingly complex to pull just the CN out of an x509 cert, but it's generally easier if the caller
+                            // has access to other libraries like bouncy castle.
+                            Optional<X509Certificate> cert = req.context().get(WebServerTls.CLIENT_X509_CERTIFICATE, X509Certificate.class);
+                            res.send(cert.map(X509Certificate::getSubjectX500Principal).map(X500Principal::getName)
+                                .map(name -> Pattern.compile("(?:^|,\s?)(?:CN=(?<val>\"(?:[^\"]|\"\")+\"|[^,]+))").matcher(name))
```
Comment (tomas-langer) [inline webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java:77]: The pattern seems to be wrong - if you want to have \s as whitespace, it must be double escaped in Java (e.g. `\\s`). Probably true for other escapes in this regexp (and same in `master` in the original PR)

**Review by tomas-langer — CHANGES_REQUESTED** [review]
See comment in `MtlsTest.java` - this fails the build

File: webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java (hunk 1 above)
Comment (astromechza) [reply webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java:77]: Looking into this now. Have been a bit busy today. It's interesting that this causes a compilation failure on the 2.x branch (java 11?) but not on the master branch (java 17?). That's why I didn't catch it there 🤔

Comment (astromechza) [reply webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java]: This is fixed now.

**Review by Verdent — APPROVED** [review]

**Pull request merged**